src/ert/ensemble_evaluator/builder/_legacy.py

```diff
@@ -54,23 +57,74 @@ def __init__(  # pylint: disable=too-many-arguments
         self._job_queue = queue_config.create_job_queue()
         self._analysis_config = analysis_config
         self._config: Optional["EvaluatorServerConfig"] = None
-        self._output_bus: "asyncio.Queue[CloudEvent]" = asyncio.Queue()
+        self._output_bus: "asyncio.Queue[com_protocol.DispatcherMessage]" = (
+            asyncio.Queue()
+        )
+
+    def generate_event_creator(
+        self, experiment_id: Optional[str] = None
+    ) -> Callable[
+        [str, Optional[int]], Union[CloudEvent, com_protocol.DispatcherMessage]
+    ]:
+        if experiment_id is not None:
+
+            def func(
+                status: str, real_id: Optional[int] = None
+            ) -> com_protocol.DispatcherMessage:
+                assert experiment_id  # mypy error
```

Comment:
strange that this is necessary when you have  `if experiment_id is not None` above

Reply:
(note that this assert would return False on an empty string, that is perhaps intentional, but probably not a mypy thing)

Reply:
> strange that this is necessary when you have `if experiment_id is not None` above

This is strange for me too, but that's how it is.